test/policies/adopter_application_policy_test.rb

```diff
@@ -0,0 +1,189 @@
+require "test_helper"
+
+# See https://actionpolicy.evilmartians.io/#/testing?id=testing-policies
+class AdopterApplicationPolicyTest < ActiveSupport::TestCase
+  include PetRescue::PolicyAssertions
+
+  context "#index?" do
+    setup do
+      @policy = -> {
+        AdopterApplicationPolicy.new(AdopterApplication, user: @user)
+      }
+      @action = -> { @policy.call.apply(:index?) }
+    end
+
+    context "when user is nil" do
+      setup do
+        @user = nil
+      end
+
+      should "return false" do
+        assert_equal @action.call, false
+      end
+    end
+
+    context "when user is adopter without adopter account" do
+      setup do
+        @user = create(:user)
+      end
+
+      should "return false" do
+        assert_equal @action.call, false
+      end
+    end
+
+    context "when user is adopter without profile" do
+      setup do
+        @user = create(:user, :adopter_without_profile)
+      end
+
+      should "return false" do
+        assert_equal @action.call, false
+      end
+    end
+
+    context "when user is adopter with profile" do
+      setup do
+        @user = create(:user, :adopter_with_profile)
+      end
+
+      should "return true" do
+        assert_equal @action.call, true
+      end
+    end
+  end
```

Comment:
Adding to above comment - adopter user should only see the applications they made.

Reply:
This is related to the other comment reply. I did add a test for `relation_scope` to test the scope in the next commit c0e41c5.

Sorry for the confusion! I should have probably amended this commit once I made these changes before making the next commit.